android/java/org/chromium/chrome/browser/settings/BackgroundImagesPreferences.java

```diff
@@ -80,6 +74,12 @@ public void onActivityCreated(Bundle savedInstanceState) {
                                     BravePref.NEW_TAB_PAGE_SHOW_SPONSORED_IMAGES_BACKGROUND_IMAGE));
             mShowSponsoredImagesPref.setOnPreferenceChangeListener(this);
         }
+        if (!NTPBackgroundImagesBridge.enableSponsoredImages()) {
+            PreferenceCategory preferenceCategory =
+                    (PreferenceCategory) findPreference(PREF_BACKGROUND_IMAGES_CATEGORY);
+            preferenceCategory.removePreference(mShowSponsoredImagesPref);
+        }
+
```

Comment:
Curious the code block in the above `if (mShowSponsoredImagesPref != null) {`
should be excuted if `enableSponsoredImages()` is false?
If not maybe they only need to be run when `enableSponsoredImages()` is `true`?

Reply:
@simonhong if we decide to use the condition when we render the pref, in that we need to cretae pref dynamically. so here instead of doing that we basically create an instance for the pref and show/hide based on the condition if that makes sense.